web/src/views/admin/AdminSettingsWrapper.vue

```diff
@@ -0,0 +1,92 @@
+<template>
+  <Scaffold v-model:active-tab="activeTab" enable-tabs disable-tab-url-hash-mode>
+    <template #title>
+      {{ $t('settings') }}
+    </template>
+    <Tab id="info" :title="$t('info')" />
+    <Tab id="secrets" :title="$t('secrets.secrets')" />
+    <Tab id="registries" :title="$t('registries.registries')" />
+    <Tab id="repos" :title="$t('admin.settings.repos.repos')" />
+    <Tab id="users" :title="$t('admin.settings.users.users')" />
+    <Tab id="orgs" :title="$t('admin.settings.orgs.orgs')" />
+    <Tab id="agents" :title="$t('admin.settings.agents.agents')" />
+    <Tab id="queue" :title="$t('admin.settings.queue.queue')" />
+
+    <router-view />
+  </Scaffold>
+</template>
+
+<script lang="ts" setup>
+import { computed, onMounted } from 'vue';
+import { useI18n } from 'vue-i18n';
+import { useRoute, useRouter } from 'vue-router';
+
+import Scaffold from '~/components/layout/scaffold/Scaffold.vue';
+import Tab from '~/components/layout/scaffold/Tab.vue';
+import useAuthentication from '~/compositions/useAuthentication';
+import useNotifications from '~/compositions/useNotifications';
+
+const notifications = useNotifications();
+const router = useRouter();
+const route = useRoute();
+const i18n = useI18n();
+const { user } = useAuthentication();
+
+onMounted(async () => {
+  if (!user?.admin) {
+    notifications.notify({ type: 'error', title: i18n.t('admin.settings.not_allowed') });
+    await router.replace({ name: 'home' });
+  }
+
+  if (!user?.admin) {
+    notifications.notify({ type: 'error', title: i18n.t('admin.settings.not_allowed') });
+    await router.replace({ name: 'home' });
+  }
+});
+
+const activeTab = computed({
+  get() {
+    if (route.name === 'admin-settings-secrets') {
```

Comment:
Wouldn't it be easier to use the route name as the tab id directly? Or we change the tab component to take a `:to=` prop, that checks if the route is active and routes on click?